torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -201,6 +171,101 @@ Reducer::~Reducer() noexcept(false) {
   }
 }
 
+// Verifies replicas in this process treat the same number of params,
+// all params require grad, and corresponding params across replicas
+// have the same dtype/size/layout.
+void Reducer::verify_replicas_within_process() {
```

Comment:
Would I be correct if I assume nothing changes here, this is justing wrapping previous code into one dedicated function?

Reply:
Correct, it makes the constructor a lot more readable, especially since now I have two pretty verbose functions that check params intra and interprocess.